### PR TITLE
OSX Build Fix

### PR DESCRIPTION
**Pull request by adam-waldenberg**
Build failing for various reasons on OS X. This PR introduces some changes to fix builds under OS X.

### CODE REVIEW

File: arthur-maven-plugin/src/main/java/org/apache/geronimo/arthur/maven/mojo/ArthurMojo.java [hunk 1]
```diff
@@ -194,8 +195,9 @@ private String buildDownloadUrl(final String graalPlatform) {
         final String graalJavaVersion = versionIncludesJavaVersion ?
                 versionSegments[versionSegments.length - 1].substring(1) :
                 System.getProperty("java.version", "1.8").startsWith("8") ? "8" : "11";
-        final String githubPlatform = graalPlatform.contains("win") ?
-                "windows-amd64" : (graalPlatform.contains("linux") ? "linux-amd64" : "darwin-amd64");
+        final String githubPlatform = graalPlatform.toLowerCase(ROOT).contains("win") ?
+                "windows-amd64" : (graalPlatform.toLowerCase(ROOT).contains("linux") ?
```
Comment (rmannibucau) [inline arthur-maven-plugin/src/main/java/org/apache/geronimo/arthur/maven/mojo/ArthurMojo.java:199]: is it an option to reverse the test to fallback on linux (as before) and not on macos?
rest looks perfect

**Comment by adam-waldenberg** [comment]
So a small note... There are actually more things that need to be done under both OSX and Windows to make Arthur work. They both need an intermediate step whete you have to do `gu install native-image`. We solved it by combining Arthur with the download and antrun plugins with some various trickery in order to make it work... At some point it would be nice to see if this could be solved in Arthur completely.

**Comment by rmannibucau** [comment]
@adam-waldenberg this is supposed to be handled by https://github.com/apache/geronimo-arthur/blob/a4241db1120e11595833ba339ad8a2610f527ef8/arthur-impl/src/main/java/org/apache/geronimo/arthur/impl/nativeimage/installer/SdkmanGraalVMInstaller.java#L84, happy to get some enhancements since intent is to work OOTB (sadly I don't have a mac handy).
Only small detail I want to keep is any fallback logic to be on linux to keep current behavior but we definitively want osx to work OOTB.

**Comment by jeanouii** [comment]
Hi,

Can you give it a try now, it looks like fixed now. Thanks for your contribution.

**Comment by jeanouii** [comment]
Merging and we can do another iteration if you still encouter issues

**Comment by adam-waldenberg** [comment]
Great.

Yes there is more to do here... We had to jump through hoops and pretty much disable much of what Arthur does oh OSX and Windows to get it to cooperate. I will see if I can take another look once I have some time over.

**Pull request merged**